apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/LoggingSnippets.cs

```diff
@@ -0,0 +1,180 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Diagnostics.Common.IntegrationTests;
+using Google.Cloud.Diagnostics.Common.Tests;
+using Google.Cloud.Logging.Type;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Snippets
+{
+    [SnippetOutputCollector]
+    public class LoggingSnippetsTests : IDisposable
+    {
+        private static readonly LogEntryPolling s_polling = new LogEntryPolling(TimeSpan.FromSeconds(120));
+
+        private readonly string _testId;
+
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        private readonly DateTime _startTime;
+
+        public LoggingSnippetsTests()
+        {
+            _testId = Utils.GetTestId();
+            IWebHostBuilder builder;
+#if NETCOREAPP2_0
+            builder = new WebHostBuilder().UseStartup<LoggingTestApplicationCore2>();
+#else
+            builder = new WebHostBuilder().UseStartup<LoggingTestApplication>();
+#endif
+            _server = new TestServer(builder);
+            _client = _server.CreateClient();
+
+            _startTime = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Test for RegisterGoogleLogger, RegisterGoogleLogger2 and UseGoogleLogger snippets.
+        /// The RegisterGoogleLogger and RegisterGoogleLogger2 snippets are configuration snippets
+        /// so the best way to test it is by checking if logging works properly.
+        /// </summary>
+        [Fact]
+        public async Task Logs_Information()
+        {
+            await _client.GetAsync($"/LoggingSamples/LogInformation/{_testId}");
+
+            PollAndVerifyLog(_startTime, _testId);
+        }
+
+        internal static void PollAndVerifyLog(DateTime startTime, string testId)
+        {
+            var results = s_polling.GetEntries(startTime, testId, 1, LogSeverity.Info);
+            results = from result in results
+                      where result.JsonPayload?.Fields["log_name"]?.StringValue?.Equals(LoggingSamplesController.LoggerName) ?? false
+                      select result;
+
+            Assert.Equal(1, results.Count());
+
+            var logEntry = results.FirstOrDefault();
```

Comment:
You can use .Single() here and skip the above

Reply:
Done